dbms/src/Columns/ColumnDecimal.cpp

```diff
@@ -56,6 +56,51 @@ int ColumnDecimal<T>::compareAt(size_t n, size_t m, const IColumn & rhs_, int) c
     return decimalLess<T>(b, a, other.scale, scale) ? 1 : (decimalLess<T>(a, b, scale, other.scale) ? -1 : 0);
 }
 
+ALWAYS_INLINE inline size_t getDecimal256BytesSize(const Decimal256 & val)
+{
+    return sizeof(bool) + sizeof(size_t) + val.value.backend().size() * sizeof(boost::multiprecision::limb_type);
+}
+
+ALWAYS_INLINE inline char * serializeDecimal256Helper(char * dst, const Decimal256 & data)
+{
+    /// deserialize Decimal256 in `Non-trivial, Binary` way, the deserialization logical is
+    /// copied from https://github.com/pingcap/boost-extra/blob/master/boost/multiprecision/cpp_int/serialize.hpp#L133
+    const auto & val = data.value.backend();
+
+    const bool s = val.sign();
+    tiflash_compiler_builtin_memcpy(dst, &s, sizeof(bool));
+    dst += sizeof(bool);
+
+    const size_t limb_count = val.size();
+    tiflash_compiler_builtin_memcpy(dst, &limb_count, sizeof(size_t));
+    dst += sizeof(size_t);
+
+    const size_t limb_size = limb_count * sizeof(boost::multiprecision::limb_type);
+    inline_memcpy(dst, val.limbs(), limb_size);
+    dst += limb_size;
+    return dst;
+}
+
+ALWAYS_INLINE inline const char * deserializeDecimal256Helper(Decimal256 & value, const char * ptr)
+{
+    auto & val = value.value.backend();
+
+    size_t offset = 0;
+    bool s = unalignedLoad<bool>(ptr + offset);
+    offset += sizeof(bool);
+    auto limb_count = unalignedLoad<size_t>(ptr + offset);
+    offset += sizeof(size_t);
+
+    val.resize(limb_count, limb_count);
+    memcpy(val.limbs(), ptr + offset, limb_count * sizeof(boost::multiprecision::limb_type));
+    if (s != val.sign())
```

Comment:
Do we have better way to set the sign flag here?

Reply:
I think it's good enough. Also in next PR I will change the serialize method suggested by @gengliqi and @windtalker . So this code will be deleted in the next PR.

Reply:
OK